### PR TITLE
Ignore metadata on schema merge

### LINKED ISSUE #936: Schma merging check is too strict
**Describe the bug**
I am attempting to read two parquet files (produced by Spark) as a single table and getting error:
```
Error during planning: The Parquet files have 2 different schemas and DataFusion does not yet support schema merging
```

Inspecting both files with `parquet-schema` shows identical schemas:
```
message spark_schema {
  required int96 system_time;
  optional int32 date (DATE);
  optional int64 usd_am (DECIMAL(18,4));
  optional int64 usd_pm (DECIMAL(18,4));
  optional int64 gbp_am (DECIMAL(18,4));
  optional int64 gbp_pm (DECIMAL(18,4));
  optional int64 euro_am (DECIMAL(18,4));
  optional int64 euro_pm (DECIMAL(18,4));
}
```

Inspecting with `parquet-meta` seems to show only one relevant difference:
* First file has `extra:       org.apache.spark.version = 3.0.1`
* Second file has `extra:       org.apache.spark.version = 3.1.2` 

**To Reproduce**
Place attached files in a directory and use `ExecutionContext::read_parquet` API.
[example.tar.gz](https://github.com/apache/arrow-datafusion/files/7036002/example.tar.gz)

**Expected behavior**
Insignificant differences in metadata are ignored and true cases that require schema merging are reported.

### PR DESCRIPTION
**Pull request by Smurphy000**
# Which issue does this PR close?

Closes #936

 # Rationale for this change

Make schema checking less strict by only comparing the fields of the schema and not the additional metadata.

### CODE REVIEW

**Comment by houqp** [comment]
cc @rdettai @yjshen

**Review by houqp — APPROVED** [review]

**Review by rdettai — APPROVED** [review]

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Looks good - maybe adding a test would make sense here?

**Comment by houqp** [comment]
@Dandandan are you aware of any existing test file that we can leverage for constructing the test here? Or should we generate the parquets inside the test?

**Comment by Smurphy000** [comment]
I tested the change using the example provided in the issue, but I was not sure if / how something like that would want to be added to the project.

**Comment by alamb** [comment]
I agree a test would be good (so that we don't accidentally break this behavior)

Perhaps we could make two simple parquet files, one with metadata and one without using the API in https://docs.rs/parquet/5.4.0/parquet/arrow/arrow_writer/index.html# ?

**Comment by alamb** [comment]
Thank you for the contribution @Smurphy000 !

**Comment by houqp** [comment]
Agree with @alamb , it would be better to create the parquet files in test on the fly if we don't have existing files that we could leverage. In the long run, we could accumulate a golden test dataset for datafusion/ballista, but this is probably out of the scope of this PR since it would require jumping through multiple hops to setup a new repo etc.

**Comment by Smurphy000** [comment]
I have now written a test where I am writing two parquet files with only differing metadata. Now when I read them back into a dataframe (where the error would occur before) it seems to always make the resulting metadata in the RecordBatches (after df.collect()) to be empty, though I expected the order of the files to determine what the resulting metadata would show. Thus I am a bit stuck as to how to confirm this does what I intended or expected.

**Comment by houqp** [comment]
I think what you have should be good enough to fix the bug. I don't think it's worth spending too much time on semantic of final metadata in this PR because that part of the code will be rewritten when we address https://github.com/apache/arrow-datafusion/issues/132. When a metadata conflicts happens, I am expecting the schema merge code to either remove only the conflicting key from the metadata or remove all metadata entirely for simplicity.

**Comment by houqp** [comment]
@Smurphy000 there are some clippy issues that need to be addressed.

**Review by yjshen — APPROVED** [review]

**Comment by alamb** [comment]
> I don't think it's worth spending too much time on semantic of final metadata 

I agree with @houqp  here -- you could also file a new issue with a link to this PR if you want to record the non-obvious behavior for further analysis

File: datafusion/src/execution/context.rs [hunk 1]
```diff
@@ -3753,6 +3753,67 @@ mod tests {
...
+                            .unwrap();
+
+                    writer.write(&rec_batch).unwrap();
+                    writer.close().unwrap();
+                }
+            }
+            _ => {}
+        }
+
+        // Read the parquet files into a dataframe to confirm results
```
Comment (alamb) [inline datafusion/src/execution/context.rs:3805]:
```suggestion
        // Read the parquet files into a dataframe to confirm results 
        // (no errors)
```

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @Smurphy000 , this is a great first contribution!

**Comment by Smurphy000** [comment]
> Thanks @Smurphy000 , this is a great first contribution!

Thank you :)
Will be keeping up to see where else I can possibly help out!

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Thanks @Smurphy000

**Comment by houqp** [comment]
Thank you @Smurphy000! Looking forward to seeing more PRs coming from you :D

**Pull request merged**